metricflow/dataflow/dataflow_plan.py

```diff
@@ -81,6 +86,12 @@ def aggregated_to_elements(self) -> Set[LinkableInstanceSpec]:
         """Indicates that the node has been aggregated to these specs, guaranteeing uniqueness in all combinations."""
         return set()
 
+    def __lt__(self, other: ComparisonAnyType) -> bool:  # noqa: D105
+        if not isinstance(other, DataflowPlanNode):
+            raise NotImplementedError
```

Comment:
Should we add an error message here in case this somehow gets hit? Seems similar to a bare `AssertionError`.

Reply:
This is actually a special value that is supposed to be returned for `__lt__` when you try to compare two objects that aren't comparable:

https://docs.python.org/3/library/constants.html